### PR TITLE
Change initial routing to namespace to be in a separate component

### PR DESCRIPTION
**Pull request by elfenheart**
Namespace dropdown still has to be refactored, but that change will be in a separate PR, along with other namespace dropdown behavioral changes

### CODE REVIEW

File: cdap-ui/app/cdap/main.js [hunk 1]
```diff
@@ -160,31 +93,25 @@ class CDAP extends Component {
...
+
           <div className="container-fluid">
-            {this.props.children}
+            <Match exactly pattern="/" component={RouteToNamespace} />
+            <Match exactly pattern="/notfound" component={Missed} />
+            <Match exactly pattern="/management" component={Management} />
+            <Match exactly pattern="/ns/:namespace" component={Home} />
+            <Match exactly pattern="/ns/:namespace/dashboard" component={Dashboard} />
+            <Match pattern="/Experimental" component={Experimental} />
+            <Match pattern="/socket-example" component={ConnectionExample} />
```
Comment (nitinmotgi) [inline cdap-ui/app/cdap/main.js:110]: Why is this socket-example ?

File: cdap-ui/app/cdap/main.js [hunk 2]
```diff
@@ -160,31 +93,25 @@ class CDAP extends Component {
...
 
-    this.namespace = Store.getState().selectedNamespace;
-    this.namespaceList = Store.getState().namespaces;
-
-    if(!this.namespace || !this.namespaceList){
-      return null;
-    }
-
     return (
       <Router basename="/cask-cdap">
```
Comment (nitinmotgi) [inline cdap-ui/app/cdap/main.js:97]: Can we change this to just cdap

File: cdap-ui/app/cdap/main.js (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/main.js:110]: this is only an example page for how to use the websocket in react, only for internal use. Will be removed

File: cdap-ui/app/cdap/components/HeaderActions/index.js [hunk 3]
```diff
@@ -20,9 +20,10 @@ import cookie from 'react-cookie';
 import {Link} from 'react-router';
 import PlusButton from '../PlusButton';
 import T from 'i18n-react';
+import orderBy from 'lodash/orderBy';
+import Store from '../../services/store/store.js';
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/HeaderActions/index.js]: Would be better if we use the `services` alias.

File: cdap-ui/app/cdap/components/RouteToNamespace/index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  findNamespace(list, name) {
+    return find(list, {name: name});
+  }
+
+  setNamespace() {
+    let list = Store.getState().namespaces;
+
+    if (!list || list.length === 0) { return;  }
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/RouteToNamespace/index.js]: nit: extra space after return?

**Review by ajainarayanan — COMMENTED** [review]
Minor comments. Otherwise 👍 LGTM.

File: cdap-ui/app/cdap/main.js (hunk 2 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/main.js:97]: Will be addressed in separate PR

File: cdap-ui/app/cdap/components/HeaderActions/index.js (hunk 3 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/HeaderActions/index.js]: Modified

File: cdap-ui/app/cdap/components/RouteToNamespace/index.js (hunk 4 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/RouteToNamespace/index.js]: removed

**Comment by ajainarayanan** [comment]
👍 🎱 Merge

**Pull request merged**